src/App.tsx

```diff
@@ -1,14 +1,70 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './App.scss';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState<string>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setIsLoaded(true);
+      });
+  }, []);
+
+  const filterTodos = (
+    allTodos:Todo[],
+    completionStatus:string,
+    searchInput: string,
+  ) => {
+    let filteredByStatusTodos = [...allTodos];
+
+    switch (completionStatus) {
+      case 'all':
+        break;
+      case 'active':
+        filteredByStatusTodos = filteredByStatusTodos.filter(todo => !todo.completed);
+        break;
+      case 'completed':
+        filteredByStatusTodos = filteredByStatusTodos.filter(todo => todo.completed);
+        break;
+      default:
+        throw new Error();
+    }
+
+    const normalizedQuery = searchInput.trim().toLowerCase();
+
+    return filteredByStatusTodos
+      .filter(todo => todo.title
+        .trim()
+        .toLowerCase()
+        .includes(normalizedQuery));
+  };
```

Comment:
1. Move it to separate helpers file
2. You can rewrite this logic using only one filter method